### PR TITLE
postgresqlPackages.jsonb_deep_sum: init at unstable-2021-12-24

### PR DESCRIPTION
**Pull request by 1000101**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1002,6 +1002,7 @@ mapAliases ({
 
   # postgresql plugins
   cstore_fdw = postgresqlPackages.cstore_fdw;
+  jsonb_deep_sum = postgresqlPackages.jsonb_deep_sum;
```
Comment (marsam) [inline pkgs/top-level/aliases.nix:1005]: This alias is not necessary, please remove it. Besides that, LGTM

Comment (marsam) [inline pkgs/top-level/aliases.nix:1005]: This alias is not necessary, please remove it. Besides that, LGTM

Comment (1000101) [reply pkgs/top-level/aliases.nix]: I'm wondering though, why is this not needed and is in place and up to date for other extensions? :thinking:

Comment (marsam) [reply pkgs/top-level/aliases.nix]: These alias are there for historical reasons. In a previous implementation of the PostgreSQL service, the extensions were available in the top level and to use them you had to something like:
`extraPlugins = [ (pkgs.postgis.override { postgresql = pkgs.postgresql_13; }) ]`
https://discourse.nixos.org/t/heads-up-upcoming-changes-to-the-postgresql-infrastructure-and-nixos-module/297/5
with the current implementation we don't really need the extensions be available at the top level.

Comment (1000101) [reply pkgs/top-level/aliases.nix]: Thanks!

**Pull request merged**